pkg/minishift/openshift/service.go

```diff
@@ -178,6 +212,39 @@ func emptyFilter(data []string) []string {
 	return res
 }
 
+// Get service content in json format
+func getService(namespace string, ip string) (map[string]string, error) {
+	serviceNodePort := make(map[string]string)
+	cmdArgText := fmt.Sprintf("get svc -o json --config=%s -n %s", systemKubeConfigPath, namespace)
+	tokens := strings.Split(cmdArgText, " ")
+	cmdName := instanceState.InstanceConfig.OcPath
+	cmdOut, err := runner.Output(cmdName, tokens...)
+	if err != nil {
+		return nil, err
+	}
+
+	if strings.Contains(byteArrayToString(cmdOut), "No resources found") {
+		return nil, nil
+	}
+
+	var data Service
+	err = json.Unmarshal(cmdOut, &data)
+	if err != nil {
+		return nil, err
+	}
+
+	for _, item := range data.Items {
+		for _, port := range item.Spec.Ports {
+			nodePort := "No node port"
+			if port.NodePort != 0 {
+				nodePort = fmt.Sprintf("http://%s:%d", ip, port.NodePort)
```

Comment:
-1, Let's not munge data together to early. A ServiceSpec (or ServiceURL as it is called atm) struct should keep the data normalised. Than I can decide at display time how I want to display the data.